packages/admin-panel/src/pages/ProfilePage.js

```diff
@@ -35,41 +48,118 @@ const SuccessMessage = styled(SmallAlert)`
   margin-bottom: 1.5rem;
 `;
 
+const STATUS = {
+  IDLE: 'idle',
+  LOADING: 'loading',
+  SUCCESS: 'success',
+  ERROR: 'error',
+  DISABLED: 'disabled',
+};
+
+const initialState = {
+  status: STATUS.IDLE,
+  errorMessage: null,
+  successMessage: null,
+  confirmModalIsOpen: false,
+  profileImage: {
+    fileId: null,
+    data: null,
+  },
+};
+
+const initReducer = user => {
+  return { ...initialState, profileImage: { data: user.profileImage, fileId: null } };
+};
+
+function reducer(state, action) {
```

Comment:
Using a local reducer keeps the ProfilePage component nice and encapsulated but also allows us to manage the complex(ish) state management outside of the component

Reply:
It is cool! I haven't seen React.useReducer before, thanks for the intro

A suggestion here would be to keep the action/reducer flow contained, and out of the component, by adding an extra layer. Something like `connectProfilePageReducer()` which does
```
const connectProfilePageReducer = () => {
  const [
    state,
    dispatch,
  ] = React.useReducer(reducer, user, initReducer);
  const actions = {
    onUpdateStart: () => dispatch({ type: 'loading' }),
    onUpdateSuccess: () => dispatch({ type: 'success', payload: 'Profile successfully updated.' }),
    onUpdateError: error => dispatch({ type: 'error', payload: error.message }),
    onFileChangeStart: () => dispatch({ type: 'disable' }),
    onFileChangeSuccess: (user, fileName) => dispatch({
      type: 'fileChange',
      payload: {
        fileId: `${user.id}-${fileName}`,
        data: Base64,
      },
    }),
    ... etc.
  };
  return [
    state,
    actions,
  ];
};
```
All of the reducer stuff could live in a neighbouring file rather than the same file as the component. This would mean the component itself would read a bit more fluently, and not risk people getting confused by the action types or whether dispatch means it's going to redux. The downside is it means you have one extra connection to make if you're trying to understand the whole flow.

Over to you!